miranda/hq/daily.py

```diff
@@ -0,0 +1,254 @@
+import csv
+import datetime as dt
+import json
+import re
+from pathlib import Path
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import numpy as np
+import pandas as pd
+import pint
+import xarray as xr
+
+
+def units():
+    """Return a unit environment."""
+    u = pint.UnitRegistry(autoconvert_offset_to_baseunit=True)
+    null = pint.Context("none")
+    u.add_context(null)
+
+    u.define("fraction = []")
+    u.define("percent = 1e-2 fraction = pct")
+    u.define("degC = kelvin; offset: 273.15 = celsius = C")
+    u.define("d = day")
+    u.define("h = hour")  # Not the Planck constant...
+    u.define("[speed] = [length] / [time]")
+
+    u.define("[precipitation] = [mass] / [length] ** 2 / [time]")
+    u.define("mmday = 1000 kg / meter ** 2 / day")
+
+    u.define("[discharge] = [length] ** 3 / [time]")
+    u.define("cms = meter ** 3 / second")
+
+    hydro = pint.Context("hydro")
+    hydro.add_transformation(
+        "[mass] / [length]**2",
+        "[length]",
+        lambda ureg, x: x / (1000 * ureg.kg / ureg.m ** 3),
+    )
+    hydro.add_transformation(
+        "[mass] / [length]**2 / [time]",
+        "[length] / [time]",
+        lambda ureg, x: x / (1000 * ureg.kg / ureg.m ** 3),
+    )
+    hydro.add_transformation(
+        "[length] / [time]",
+        "[mass] / [length]**2 / [time]",
+        lambda ureg, x: x * (1000 * ureg.kg / ureg.m ** 3),
+    )
+    u.add_context(hydro)
+    u.enable_contexts(hydro)
+    return u
+
+
+unites = units()
+
+# CMOR-like attributes
+cmor = json.load(open(Path(__file__).parent / "cf_attrs.json"))["variable_entry"]
```

Comment:
Is the CMOR3 library something that we could leverage here to reduce the complexity? https://cmor.llnl.gov/mydoc_cmor3_python/

Reply:
Unclear. Based on a very quick assessment, it will come with the CMIP CVs, which assume metadata for model output. I think we probably want more flexibility.

Reply:
I see, it's very specifically catered to CMIP and not so much to the indices/variables that HQ is using. That makes sense. Just curious.